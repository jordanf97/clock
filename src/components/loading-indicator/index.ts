export { LoadingIndicator } from "./loading-indicator";
export { OverlayLoadingIndicator } from "./overlay-loading-indicator";
