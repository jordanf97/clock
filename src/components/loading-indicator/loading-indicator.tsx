import React from "react";
import { View } from "react-native";
import { Spinner, useTheme } from "@ui-kitten/components";
import { themeElevations } from "@/theme";
import { themeSpacing } from "@/theme/theme-spacings";

/**
 * Simple wrapper around Kitten UI spinner for
 * representing loading states
 *
 * TODO: implement normalisation on spacings + consider theme constraints
 */
export const LoadingIndicator: React.FC = () => {
  const theme = useTheme();

  return (
    <View
      style={{
        borderRadius: themeSpacing(1, "height"),
        paddingHorizontal: themeSpacing(3, "width"),
        paddingVertical: themeSpacing(3, "height"),
        backgroundColor: theme["custom-background"],
        ...themeElevations.DP6,
      }}
    >
      <Spinner status="primary" />
    </View>
  );
};
