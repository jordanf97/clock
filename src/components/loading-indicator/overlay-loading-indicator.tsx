import { CentredView } from "@/layouts";
import React from "react";
import { StyleSheet, View } from "react-native";
import { LoadingIndicator as _LoadingIndicator } from "./loading-indicator";

/**
 * Simple component responsible for covering entire screen with an overlay & loader
 */
export const OverlayLoadingIndicator: React.FC<any> = ({
  LoadingIndicator = _LoadingIndicator,
}) => (
  <View style={StyleSheet.absoluteFillObject}>
    <CentredView
      style={{
        backgroundColor: "rgba(0, 0, 0, 0.5)",
      }}
    >
      <LoadingIndicator />
    </CentredView>
  </View>
);
