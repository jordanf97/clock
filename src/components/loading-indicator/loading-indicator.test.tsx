import { render } from "@testing-library/react-native";
import { LoadingIndicator } from "./loading-indicator";
import React from "react";

it("renders correctly", () => {
  const tree = render(<LoadingIndicator />).toJSON();

  expect(tree).toMatchSnapshot();
});
