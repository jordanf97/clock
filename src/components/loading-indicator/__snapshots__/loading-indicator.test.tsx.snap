// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "theme-key-custom-background",
      "borderRadius": 8,
      "elevation": 6,
      "paddingHorizontal": 32,
      "paddingVertical": 32,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 3,
        "width": 0,
      },
      "shadowOpacity": 0.27,
      "shadowRadius": 4.65,
    }
  }
/>
`;
